ext/sodium/sodium_pwhash.c

```diff
@@ -40,9 +40,35 @@
 #define PHP_SODIUM_PWHASH_OPSLIMIT 4
 #define PHP_SODIUM_PWHASH_THREADS 1
 
+static inline int get_options(zend_array *options, size_t *memlimit, size_t *opslimit) {
+	zval *opt;
```

Comment:
Possibly also handle the default value assignment? I.e. add

```
*opslimit = PHP_SODIUM_PWHASH_OPSLIMIT;
*memlimit = PHP_SODIUM_PWHASH_MEMLIMIT << 10;
if (!options) {
    return SUCCESS;
}
```

here and then make the call just

```
size_t memlimit, opslimit;
if (get_options(options, &memlimit, &opslimit) == FAILURE) {
    // ...
}
```

?